pkg/registry/apis/provisioning/resources/parser.go

```diff
@@ -26,22 +26,25 @@ import (
 var ErrNamespaceMismatch = errors.New("the file namespace does not match target namespace")
 
 type FileParser struct {
-	// The target repository
-	repo repository.Repository
-
 	// client helper (for this namespace?)
-	client *DynamicClient
-	kinds  KindsLookup
+	client     *DynamicClient
+	kinds      KindsLookup
+	repository *provisioning.Repository
+	linter     lint.Linter
 }
 
-func NewParser(repo repository.Repository, client *DynamicClient, kinds KindsLookup) *FileParser {
+func NewParser(repository *provisioning.Repository, client *DynamicClient, kinds KindsLookup) *FileParser {
```

Comment:
I like this change -- just use the config, not the instance; but it makes the future plan (IIUC) of reading lint config from the repository itself more complicated

Keeping full access to the repository means we can read the config file and create the linter